### PR TITLE
Replaced switch-cases in strange place for it

### PR DESCRIPTION
**Pull request by JeanPaulLucien**
## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Type: none
Notes: small code's refactoring.
Fixes: it was strange place for switch-case
**Visual changes:** no. It's same.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/HtmlUtils.tsx [hunk 1]
```diff
@@ -731,31 +731,18 @@ export function linkifyAndSanitizeHtml(dirtyHtml: string, options = linkifyMatri
...
-        case "DIV":
-            // don't treat math nodes as block nodes for deserializing
-            return !(node as HTMLElement).hasAttribute("data-mx-maths");
-        default:
-            return false;
+    if (node.nodeName == "DIV") {
+        // don't treat math nodes as block nodes for deserializing
+        return !(node as HTMLElement).hasAttribute("data-mx-maths");
     }
+    else {
```
Comment (weeman1337) [inline src/HtmlUtils.tsx:738]: If there is a return inside the `if`-block, you can remove the `else`.

File: src/HtmlUtils.tsx [hunk 2]
```diff
@@ -731,31 +731,18 @@ export function linkifyAndSanitizeHtml(dirtyHtml: string, options = linkifyMatri
...
+        let trueNodeNames = [2, 21, "H1", "H2", "H3", "H4", "H5", "H6", "PRE", "BLOCKQUOTE", "P", "UL", "OL", "LI", "HR", "TABLE", "THEAD", "TBODY", "TR", "TH", "TD"];
+        do {
```
Comment (weeman1337) [inline src/HtmlUtils.tsx:740]: This looks really inefficient. I would suggest to store `trueNodeNames` outside of the function and then test by `trueNodeNames.includes` for it. Alternatively you can use a `Map` and call `has` on it.

The entire `do` loop can then be replaced with →

```
return trueNodeNames.includes(node.nodeName);
```

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Thank you for your contribution @JeanPaulLucien 

Please have a look at my comments. Also be sure to follow [the code style](https://github.com/vector-im/element-web/blob/develop/code_style.md).

File: src/HtmlUtils.tsx [hunk 3]
```diff
@@ -731,31 +731,16 @@ export function linkifyAndSanitizeHtml(dirtyHtml: string, options = linkifyMatri
...
+        return !(node as HTMLElement).hasAttribute("data-mx-maths");
     }
+    let okNodeNames = [2, 21, "H1", "H2", "H3", "H4", "H5", "H6", "PRE", "BLOCKQUOTE", "P", "UL", "OL", "LI", "HR", "TABLE", "THEAD", "TBODY", "TR", "TH", "TD"];
+    do {
+        if (node.nodeName == okNodeNames[okNodeNames[0]]) {
+            return true;
+        }
+        okNodeNames[0] = okNodeNames[0] + 1;
+    } while (okNodeNames[0] < okNodeNames[1]);
+    return false;
```
Comment (justjanne) [inline src/HtmlUtils.tsx:745]:
```suggestion
    return ["H1", "H2", "H3", "H4", "H5", "H6", "PRE", "BLOCKQUOTE", "P", "UL", "OL", "LI", "HR", "TABLE", "THEAD", "TBODY", "TR", "TH", "TD"].includes(node.nodeName);
```

**Review by justjanne — CHANGES_REQUESTED** [review]

File: src/HtmlUtils.tsx (hunk 3 above)
Comment (JeanPaulLucien) [reply src/HtmlUtils.tsx:745]: `return false;` was in original `switch-case`. I'm not sure what will going on without `return false;`.

File: src/HtmlUtils.tsx (hunk 2 above)
Comment (weeman1337) [reply src/HtmlUtils.tsx]: Element Web does only support the two latest major browser versions, see https://github.com/vector-im/element-web#supported-environments.

This is also transpiled so you don't have to worry about it.

A single map defined **outside** of the function can be an improvement. Create once with a good lookup time.

File: src/HtmlUtils.tsx (hunk 3 above)
Comment (JeanPaulLucien) [reply src/HtmlUtils.tsx:745]: I need an optimization. I will not use `.includes` here.

**Comment by JeanPaulLucien** [comment]
Will care about `loop + if` vs `.includes` in next time.

**Comment by weeman1337** [comment]
Hello @JeanPaulLucien :wave: A little reminder if you can correct the code style here. Then we would be ready to go. Once done please re-request review.

image

**Comment by weeman1337** [comment]
I missed one point. [Please sign-off your changes according to the contribution guideline](https://github.com/vector-im/element-web/blob/develop/CONTRIBUTING.md#sign-off).

**Comment by justjanne** [comment]
As there is no good solution for this that supports dark and light mode users equally well, we won't be accepting this contribution at this time.

**Comment by justjanne** [comment]
As this currently does not improve the readability or code style in any meaningful way and the discussion has taken significant amounts of time already, we won't be accepting this contribution at this time.

**Pull request closed without merging**